session.go

```diff
@@ -151,7 +150,15 @@ func (s *session) recvLoop() {
 	stoppedOnExpectedEOF := false
 
 	defer func() {
+		log.Debug("Closing lampshade TCP connection")
+		if s.requiredSessions != nil {
+			log.Debugf("Requesting new session: %#v", s.requiredSession)
+			s.requiredSessions <- s.requiredSession
+		}
 		closeErr := s.Conn.Close()
+		if s.lifecycle != nil {
```

Comment:
Not a bit deal, but as we already have `NoopClientLifecycleListener`, maybe pass it when start session in `listener.go` too, so this check can be removed.

Reply:
And potentially avoid panic in other places too

Reply:
good idea -- will do!